src/main/scala/com/fulcrumgenomics/fastq/FastqToBam.scala

```diff
@@ -72,6 +71,13 @@ import htsjdk.samtools.{ReservedTagConstants, SAMFileHeader, SAMReadGroupRecord}
     |For more information on read structures see the
     |[Read Structure Wiki Page](https://github.com/fulcrumgenomics/fgbio/wiki/Read-Structures)
     |
+    |UMIs may be extracted from the read sequences, the read names, or both.  If `--extract-umis-from-read-names` is
+    |specified, any UMIs present in the read names are extracted; read names are expected to be `:`-separated with
+    |any UMIs present in the 8th field.  If this option is specified, the `--umi-qual-tag` option may not be used as
+    |qualities are not available for UMIs in the read name. If UMI segments are present in the read structures those
+    |will also be extracted.  If UMIs are present in both, the final UMIs are constructed by first taking the UMIs
+    |from the read names, then adding a hyphen, then the UMIs extracted from the reads.
```

Comment:
I really hope we don't see UMIs in the read name and the reads themselves.  The reason they're in names to begin with is someone extracted them to begin with.  So I don't think we should support UMIs in both read names and in the reads themselves, even though it's easy enough to do.

Reply:
My thought process here was 1) it's easy enough to do and 2) if it's possible then sooner or later someone will do it.  Illumina being illumina only supports UMIs in ways _their_ library preps put UMIs onto molecules.  What if someone uses two UMIs with one being in a place Illumina supports (e.g. inline) and the other being in a place they don't (e.g. inline after a skip, or as the second half of an index read)?